### PR TITLE
Icons for postcss configs

### PR DESCRIPTION
**Pull request by r3nya**
PostCSS core has a config file for plugin.
It can be [`.postcssrc`](https://github.com/michael-ciniawsky/postcss-load-config#postcssrc) (like json file) and [`postcss.config.js`](https://github.com/michael-ciniawsky/postcss-load-config#postcssconfigjs) (like js file).

I update rule for PostCSS file, please look.

### CODE REVIEW

File: src/icons.json [hunk 1]
```diff
@@ -621,6 +621,22 @@
...
+        "name": "Config file for Post​CSS",
+        "extensions": [
+          ".postcssrc"
+        ],
+        "base": "source.json",
+        "scope": "source.json.settings"
+      },
+      {
+        "name": "JS Config file for Post​CSS",
+        "extensions": [
```
Comment (oivva) [inline src/icons.json]: Remove this section `JS Config file for Post​CSS` if extension is provided we always try to use default js icon. There is only some exceptions such as `gulpfile.js` or `package.json`. See #5 for more details.

Comment (oivva) [inline src/icons.json]: `Config File for Post​CSS` is an alias, so place it to the `aliases` not the `syntaxes`. Also change the scope `source.json.settings` to `source.json.postcss`, `source.json.settings` is used for `gear` icon.

**Comment by oivva** [comment]
Thanks a lot for your pull request! Фsk me if anything is unclear :wink:

**Comment by r3nya** [comment]
@oivva need I build it? I mean run `npm run build`?

**Comment by r3nya** [comment]
Btw, please review my last commit.

File: dist/preferences/file_type_postcss.tmPreferences [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <plist version="1.0">
   <dict>
     <key>scope</key>
-    <string>source.postcss</string>
+    <string>source.postcss, source.json.postcss, source.js.settings</string>
```
Comment (r3nya) [inline dist/preferences/file_type_postcss.tmPreferences]: Understood. One moment…

**Comment by oivva** [comment]
Yes, you need build it just for check if it looks good. However do not add those changes to the commit.

**Comment by oivva** [comment]
See my latest commits

**Comment by r3nya** [comment]
@oivva fixed it.

**Comment by oivva** [comment]
Remove all this stuff, not just the name.

```js
,
       {
          "name": "JS Config file for Post​CSS",
          "extensions": [
            "postcss.config.js"
          ],
         "base": "source.js",
         "scope": "source.js.settings"
       }
```

**Comment by r3nya** [comment]
Btw, about commit messages – you can squash my commits and set the correct commit message.

**Comment by r3nya** [comment]
@oivva sorry, but I need to change icon for `postcss.config.js` and `.postcssrc` file.

Will it ok?

```json
"file_type_postcss": {
    "color": "red",
    "syntaxes": [
      {
        "name": "Syntax Highlighting for Post​CSS",
        "scope": "source.postcss"
      }
    ],
    "aliases": [
      {
        "name": "Config file for Post​CSS",
        "extensions": [
          ".postcssrc",
          "postcss.config.js"
        ],
        "base": "source.json",
        "scope": "source.json.postcss"
      }
    ]
  },
```

**Comment by oivva** [comment]
Ok, 1 min :wink:

**Comment by oivva** [comment]
"file_type_postcss": {
    "color": "red",
    "syntaxes": [
      {
        "name": "Syntax Highlighting for Post​CSS",
        "scope": "source.postcss"
      }
    ],
    "aliases": [
      {
        "name": "JSON (Post​CSS)",
        "extensions": [
          ".postcssrc"
        ],
        "base": "source.json",
        "scope": "source.json.postcss"
      },
      {
        "name": "JavaScript (Post​CSS)",
        "extensions": [
          "postcss.config.js"
        ],
        "base": "source.js",
        "scope": "source.js.postcss"
      }
    ]
  },

**Comment by r3nya** [comment]
@oivva hooray! Let's merge it! ;)

**Comment by oivva** [comment]
In simple terms, if syntax for language exists in `Package Control`, we add `syntaxes` if not we create `aliases`. 

[This code](https://github.com/oivva/st-file-icons/pull/27#issuecomment-259371392) associates icons with syntaxes and aliases.

**Comment by oivva** [comment]
Thanks I'll merge it in 5 min.

**Comment by r3nya** [comment]
@oivva Cool! I'm happy about it. 😉

**Comment by oivva** [comment]
However I'll release it near the end of the month so u can replace package files in your `Packages\zz File Icons` folder manually :wink:

**Pull request merged**